src/crypto.ts

```diff
@@ -25,8 +29,12 @@ export function generateTokenAndSharedSecret (seed: Buffer): { token: Buffer, sh
 }
 
 export function generateSharedSecretFromToken (seed: Buffer, token: Buffer): Buffer {
-  const keygen = hmac(seed, SHARED_SECRET_GENERATION_STRING)
-  const sharedSecret = hmac(keygen, token)
+  let sharedSecret = sharedSecretMap.get(token)
```

Comment:
This looks like a memory leak; the sharedSecretMap isn't being cleaned up anywhere. Because there's an entry created in each map for each shared secret, it will leak fairly rapidly (once per SPSP query). We could set a timer and allow it to regenerate the HMAC if it needs it again after a timeout, but I think the best thing to do would be to instantiate this as a class inside of a connection so that it can be garbage collected when the connection is no longer used

Reply:
Since the shared secret is per-connection, we should just store the secret on the connection object. We shouldn't need this map

Reply:
Its already stored on the connection, I thought it was being generated a ton in index.ts but looking at it more its only happening in the case where the connection does not exist. I updated this PR to remove it.

I do wonder if we still could have a memory leak with the other maps and if we should instead do a crypto class like Ben mentioned. As is I think crypto.ts will stick around even after the connection is closed so these would slowly build up. 

The only challenge with that is then we have to pass that instantiated class into the Packet for its serialize and deserialize calls. @emschwartz Curious on your thoughts.